subsys/net/lib/aws_iot/src/aws_iot.c

```diff
@@ -113,6 +116,10 @@ static char payload_buf[CONFIG_AWS_IOT_MQTT_PAYLOAD_BUFFER_LEN];
 
 static aws_iot_evt_handler_t module_evt_handler;
 
+#if defined(CONFIG_AWS_IOT_BROKER_HOST_NAME_APP)
+static char *aws_host_name;
```

Comment:
This could be `static const char*`. 

I also wonder, perhaps it'd make sense to define the variable in both cases, and initialize it with `CONFIG_AWS_IOT_BROKER_HOST_NAME` if this configuration is used? That way we could get rid of some of the ifdef slicing below.

Reply:
Agree

Reply:
I have made it into a `char` array instead, to be more in line with the way `aws_iot_config`'s `client_id` is handled including the length of the string. As suggested by @simensrostad in another comment. Maybe this is a better solution?